src/components/Table/Table.jsx

```diff
@@ -399,75 +391,21 @@ const Table = props => {
-              {options.hasRowExpansion ? <TableExpandHeader /> : null}
-              {options.hasRowSelection ? (
-                <TableHeader
-                  style={Object.assign(
-                    { paddingBottom: '0.5rem' },
-                    filterBarActive === true ? filterBarActiveStyle : {}
-                  )}>
-                  {/* TODO: Replace checkbox with TableSelectAll component when onChange bug is fixed
-                    https://github.com/IBM/carbon-components-react/issues/1088 */}
-                  <Checkbox
-                    id="select-all"
-                    labelText="Select All"
-                    hideLabel
-                    indeterminate={view.table.isSelectAllIndeterminate}
-                    checked={view.table.isSelectAllSelected}
-                    onChange={() => actions.table.onSelectAll(!view.table.isSelectAllSelected)}
-                  />
-                </TableHeader>
-              ) : null}
-              {visibleColumns.map(column => {
-                const hasSort =
-                  view.table && view.table.sort && view.table.sort.columnId === column.id;
-                return (
-                  <TableHeader
-                    id={`${id}-Header-Column-${column.id}`}
-                    key={`column-${column.id}`}
-                    style={filterBarActive === true ? filterBarActiveStyle : {}}
-                    isSortable={column.isSortable}
-                    isSortHeader={hasSort}
-                    onClick={() => {
-                      if (column.isSortable && actions.table.onChangeSort) {
-                        actions.table.onChangeSort(column.id);
-                      }
-                    }}
-                    sortDirection={hasSort ? view.table.sort.direction : 'NONE'}>
-                    {column.name}
-                  </TableHeader>
-                );
-              })}
-              {options.hasRowActions ? <TableHeader>&nbsp;</TableHeader> : null}
-            </TableRow>
-            {filterBarActive && (
-              <FilterHeaderRow
-                columns={visibleColumns.map(column => ({
-                  ...column.filter,
-                  id: column.id,
-                  isFilterable: !isNil(column.filter),
-                }))}
-                key={JSON.stringify(view.filters)}
-                filters={view.filters}
-                tableOptions={options}
-                onApplyFilter={actions.toolbar.onApplyFilter}
-              />
-            )}
-            {columnBarActive && (
-              <ColumnHeaderRow
-                columns={columns.map(column => ({
-                  id: column.id,
-                  name: column.name,
-                }))}
-                ordering={view.table.ordering}
-                tableOptions={options}
-                onChangeOrdering={actions.table.onChangeOrdering}
-              />
-            )}
-          </TableHead>
-
+          <TableHead
+            options={options}
```

Comment:
My preference is to never pass props down that aren't specified in the child component's propTypes, to make code more traceable/readable... since we only are utilizing `hasRowExpansion` and `hasRowSelection` can we pass these explicitly?

```
options={{
  hasRowSelection: options.hasRowSelection,
  hasRowExpansion: options.hasRowExpansion,
}}
```

Reply:
My bad -- I see that the `options` prop is passed directly to FilterHeaderRow / ColumnSelectRow.  in that case, how about we externalize the PropTypes shape for options into an interface file?

```
TableOptionProps = {
  hasRowSelection: PropTypes.bool,
  ...
}
```